pkg/account/component.go

```diff
@@ -541,3 +543,44 @@ func (c *component) executeActions(ctx context.Context, actions []string) error
 
 	return err
 }
+
+func (c *component) CancelEmailChange(ctx context.Context) error {
+	return c.removeStringAttributeFromUser(ctx, constants.AttrbEmailToValidate)
+}
+
+func (c *component) CancelPhoneNumberChange(ctx context.Context) error {
+	return c.removeStringAttributeFromUser(ctx, constants.AttrbPhoneNumberToValidate)
+}
+
+func (c *component) removeStringAttributeFromUser(ctx context.Context, attr kc.AttributeKey) error {
+	accessToken := ctx.Value(cs.CtContextAccessToken).(string)
+	realm := ctx.Value(cs.CtContextRealm).(string)
+
+	// Get the user representation from Keycloak
+	userKc, err := c.keycloakAccountClient.GetAccount(accessToken, realm)
+	if err != nil {
+		c.logger.Warn(ctx, "msg", "Can't get account", "err", err.Error())
+		return err
+	}
+	keycloakb.ConvertLegacyAttribute(&userKc)
+
+	if userKc.GetAttributeString(attr) == nil {
+		// Attribute is already missing, no reason to update the user
+		return nil
+	}
+
+	userKc.RemoveAttribute(attr)
+
+	// Update keycloak account
+	err = c.keycloakAccountClient.UpdateAccount(accessToken, realm, userKc)
+	if err != nil {
+		c.logger.Warn(ctx, "msg", "Can't update account", "err", err.Error())
+		return err
+	}
+
+	userID := ctx.Value(cs.CtContextUserID).(string)
```

Comment:
Are we sure to have them in the context? (perhaps populated when checking JWT)
I would just use values from userKc...

Reply:
Yes we are sure. It is the way to retrieve userID in the account api. We do even like this in the update endpoint.

We go through a middleware MakeHTTPOIDCTokenValidationMW that validates the token and put some values in the context.

Reply:
Ok... Taking them from userKc is just a direct access to the values